DecipherNow/gm-ui-components
# fix: peer dependency issue

- moved `react`, `react-dom`, `styled-components` to webpack externals and peerDependencies
- moved all dependencies except `polished` to dev dependencies
- updated `styled-components` to 3.2.6
- released `gm-ui-components@0.4.6`

I believe we tried the first two things already when we worked on this last time, since that is exactly what is recommended in the styled-components docs. I found [this issue](https://github.com/styled-components/styled-components/issues/1563) where people were getting a similar error even though they also used webpack externals and peerDeps. They only got it to work when they downgraded styled-components to 2.4.x or upgraded to 3.2.x. 🤷‍♀️ 

I tested this in the dashboard, decipher-scripts, and had @knoebelja install in his project as well. Everything looks good. 🤞 

To test: 
- update to `gm-ui-components@0.4.6` in decipher-scripts and import a component
- install `gm-ui-components@0.4.6` in the dashboard, try with older versions of `styled-components` and confirm it doesn't break
